ui/src/app/base/types.ts

```diff
@@ -67,3 +70,23 @@ export type CommonActionFormProps<E = null> = {
   processingCount: number;
   viewingDetails: boolean;
 };
+
+export type ActionStatuses = ValueOf<typeof ACTION_STATUS>;
+
+export type ModelAction<PK> = {
+  [ACTION_STATUS.error]: PK[];
+  [ACTION_STATUS.loading]: PK[];
+  [ACTION_STATUS.success]: PK[];
+};
+
+export type PayloadActionWithMeta<M, P = null> = PayloadAction<
+  P,
+  string,
+  { modelPK: M }
```

Comment:
Some methods do not use the PK of model type e.g. when we [fetch event records](https://github.com/canonical-web-and-design/maas-ui/blob/master/ui/src/app/store/event/slice.ts#L20) it uses a node id, not an event record id so we couldn't store that request in `modelActions` otherwise the object would be keyed by a mix of node and event record ids (which could also potentially be conflicting).